File: Inheritance/Payment.ts
abstract class Payment {
    abstract pay(amount: number): void;

    printReceipt(): void {
        console.log("Payment completed");
    }

}

class CardPayment extends Payment {
    pay(amount: number): void {
        console.log(`Paid ${amount} using card`);
    }
}

class UpiPayment extends Payment {
    pay(amount: number): void {
        console.log(`Paid ${amount} using UPI`);
    }
}

File: Inheritance/Payment.java
package Inheritance;

abstract class Payment {
    abstract void pay(double amount);

    void printReceipt() {
        System.out.println("Payment completed");
    }
}

class CardPayment extends Payment {
    void pay(double amount) {
        System.out.println("Paid " + amount + " using card");
    }
}

class UpiPayment extends Payment {
    void pay(double amount) {
        System.out.println("Paid " + amount + " using UPI");
    }
}
